pkg/task/task.go

```diff
@@ -0,0 +1,68 @@
+package task
+
+import "time"
+
+// TaskState (kind: int) represents the last known state of a task.
+// A task can be in one of three states
+// TaskStateScheduled: this is the initial state of the task when it enters the queue.
+// TaskStateProcessing: once work begins on the task, it is put into this state.
+// TaskStateComplete: work is no longer being done on this task. client should check task result.
+type TaskState int
+
+const (
+	StateScheduled TaskState = iota
+	StateProcessing
+	StateComplete
+)
+
+func (t TaskState) String() string {
+	return [...]string{
+		"StateRequested",
+		"StateProcessing",
+		"StateComplete",
+	}[t]
+}
+
+// TaskResultStatus (kind: int) is a status code for completed tasks.
+// TaskResultNone: initial status, No status, probably becasue the task is incomplete.
+// TaskResultSuccess: the task has completed without an error.
+// TaskResultFail: the task has completed with a failure.
+// TaskResultAbort: testground encountered an error and the task has not been scheduled.
+// testground will respond to client with a TaskResultAbort status if it is unable to enqueue
+// the reqeusted task.
+type TaskResultStatus int
+
+const (
+	ResultNone TaskResultStatus = iota
+	ResultSuccess
+	ResultFail
+	ResultAbort
```

Comment:
I am not sure why we need `ResultAbort` - at least at the moment I don't think we have such state on the Kubernetes runner.

Reply:
I was trying to indicate a difference between failures of the testground service and the status of the plan run itself. I was imaging this as "Fail" -- the plan failed, as indicated as the final result of the plan run, and "Abort" to mean testground tried to run this task and couldn't do it for some reason.